nimx/outline_view.nim

```diff
@@ -111,16 +116,28 @@ proc nodeAtIndexPath(v: OutlineView, indexPath: openarray[int]): ItemNode =
     for i in indexPath:
         result = result.children[i]
 
-proc getExpandedRowsCount(node: ItemNode): int =
-    result = node.children.len
+proc selectedNode(v: OutlineView): ItemNode =
+    v.nodeAtIndexPath(v.selectedIndexPath)
+
+proc getExposedRowsCount(node: ItemNode): int =
+    result = 1
+    if node.expanded:
+        for child in node.children:
+            result += child.getExposedRowsCount()
+
+proc getExposingRowNum(v: OutlineView, indexPath: seq[int]): int =
+    if indexPath.len == 0:
+        return -1
 
-    for i in node.children:
-        if i.expanded:
-            result += i.getExpandedRowsCount()
+    result = v.getExposingRowNum(indexPath[0..^2])
```

Comment:
Slicing a seq results in allocation + copying. Pls do your best to reduce the number of those.